### PR TITLE
[DOCS] Remove redundant connector APIs

### PR DESCRIPTION
**Pull request by lcawl**
Relates to https://github.com/elastic/kibana/issues/139763

This PR removes the following pages from the Security Guide, since they duplicate information that exists in the Kibana Guide:

- https://www.elastic.co/guide/en/security/master/register-connector.html
- https://www.elastic.co/guide/en/security/master/update-connector.html
- https://www.elastic.co/guide/en/security/master/cases-actions-api-execute.html

### CODE REVIEW

File: docs/cases/api/cases-api/cases-api.asciidoc [hunk 1]
```diff
@@ -6,8 +6,6 @@ You can create, manage, configure, and send cases to external systems with these
 
 NOTE: Cases return a warning header for deprecated endpoints. The value of the warning header is in the form `299 Kibana-{kibana_version} "{warning_text}"`
 
-* {kibana-ref}/cases-api.html[Cases API]: Used to open and manage security action items.
+* {kibana-ref}/cases-api.html[Cases APIs]: Used to open and manage security action items.
```
Comment (nastasha-solomon) [inline docs/cases/api/cases-api/cases-api.asciidoc:9]: "Action items" sounds jargony. I'm not sure what a less business-like alternative would be. I'd suggest "cases", but it's still pretty specific to Elastic/Security and it's not all that descriptive so I don't think it's a good alternative.

**Review by nastasha-solomon — APPROVED** [review]
Just one comment! Otherwise, LGTM!

File: docs/cases/api/actions-api/cases-actions-api-intro.asciidoc [hunk 2]
```diff
@@ -10,8 +10,8 @@ You can push {es-sec} cases to these third-party systems:
 * {swimlane}
 
 
-To push cases, you need to create a connector using the {kib} Actions API,
-which stores the information required to interface with the external system.
+To push cases, you need to create a connector, which stores the information
+required to interface with the external system.
```
Comment (nastasha-solomon) [inline docs/cases/api/actions-api/cases-actions-api-intro.asciidoc:14]: Oops, had one more comment. "Interface" sounds a bit overly-formal to me. How about "communicate" or "interact"?

Comment (lcawl) [inline docs/cases/api/actions-api/cases-actions-api-intro.asciidoc:14]:
```suggestion
required to communicate with the external system.
```

File: docs/cases/api/cases-api/cases-api.asciidoc (hunk 1 above)
Comment (lcawl) [reply docs/cases/api/cases-api/cases-api.asciidoc:9]: I'll leave it as it is for now but if you think of a better term, that's fine by me!

File: docs/cases/api/actions-api/cases-actions-api-intro.asciidoc (hunk 2 above)
Comment (lcawl) [reply docs/cases/api/actions-api/cases-actions-api-intro.asciidoc]: Sure!  I've changed it in 9e2c54358275ecfb5b6a120b23d685a091151114

File: docs/cases/api/actions-api/cases-actions-api-intro.asciidoc [hunk 3]
```diff
@@ -25,594 +25,7 @@ IMPORTANT: {sn} ITSM and SecOps connectors created in {stack} version 7.15.0 or
 
 To send cases to an external system and keep the {es-sec-ui} updated:
 
-. <<register-connector>>: Create the connector (Actions API).
+. {kibana-ref}/create-connector-api.html: Create the connector.
```
Comment (lcawl) [inline docs/cases/api/actions-api/cases-actions-api-intro.asciidoc:28]:
```suggestion
. {kibana-ref}/create-connector-api.html[Create connector]: Create the connector.
```

**Pull request merged**